packages/shylu/shylu_dd/frosch/src/SchwarzOperators/FROSch_IPOUHarmonicCoarseOperator_def.hpp

```diff
@@ -218,16 +218,36 @@ namespace FROSch {
                                                                   ConstXMultiVectorPtrVecPtr nodeListVec)
     {
         FROSCH_DETAILTIMER_START_LEVELID(buildCoarseSpaceTime,"IPOUHarmonicCoarseOperator::buildCoarseSpace");
+
+        this->NumberOfBlocks_ = repeatedNodesMapVec.size();
+
+        FROSCH_ASSERT(dofsPerNodeVec.size()==this->NumberOfBlocks_,"dofsPerNodeVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(repeatedDofMapsVec.size()==this->NumberOfBlocks_,"repeatedDofMapsVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(nullSpaceBasisVec.size()==this->NumberOfBlocks_,"nullSpaceBasisVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(dirichletBoundaryDofsVec.size()==this->NumberOfBlocks_,"dirichletBoundaryDofsVec.size()!=this->NumberOfBlocks_");
+        FROSCH_ASSERT(nodeListVec.size()==this->NumberOfBlocks_,"nodeListVec.size()!=this->NumberOfBlocks_");
+
         // Das könnte man noch ändern
         // TODO: DAS SOLLTE ALLES IN EINE FUNKTION IN HARMONICCOARSEOPERATOR
-        for (UN i=0; i<repeatedNodesMapVec.size(); i++) {
+        for (UN i=0; i<this->NumberOfBlocks_; i++) {
```

Comment:
Help us folks with no German knowledge in the comments please :) .. It is pretty cool Trilinos has comments in several languages though

Reply:
Haha, you are right. I should pay more attention to this. I translated this now :-)